File: AIMS/src/main/java/com/example/aims/product/models/Product.java
package com.example.aims.product.models;

public class Product {
    private String id;
    private String title;
    private String category;
    private double value;
    private double price;
    private String description;
    private int quantity;
    private String barcode;
    private String condition; // e.g., new, used
    private String dimensions;
    private double weight;

    public Product(String id, String title, String category, double value, double price, String description, int quantity, String barcode, String condition, String dimensions, double weight) {
        this.id = id;
        this.title = title;
        this.category = category;
        this.value = value;
        this.price = price;
        this.description = description;
        this.quantity = quantity;
        this.barcode = barcode;
        this.condition = condition;
        this.dimensions = dimensions;
        this.weight = weight;
    }

    public double getPrice() {
        return this.price;
    }
}

File: AIMS/Models/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AIMS.Models.Entities
{
    public class Product
    {
        public int ProductID { get; set; }
        public string Title { get; set; }
        public string Category { get; set; }
        public double Value { get; set; }
        public double Price { get; set; }
        public string Description { get; set; }
        public string Barcode { get; set; }
        public int Quantity { get; set; }
        public DateTime WarehouseEntryDate { get; set; }
        public string Dimensions { get; set; }
        public double Weight { get; set; }

        // Thuộc tính riêng cho từng loại
        public string Authors { get; set; } // Book
        public string CoverType { get; set; } // Book
        public string Publisher { get; set; } // Book
        public DateTime PublicationDate { get; set; } // Book
        public int Pages { get; set; } // Book
        public string BookGenre { get; set; } // Book

        public string Artist { get; set; } // CD, LP
        public string RecordLabel { get; set; } // CD, LP
        public string Tracklist { get; set; } // CD, LP
        public string MusicGenre { get; set; } // CD, LP
        public DateTime ReleaseDate { get; set; } // CD, LP, DVD

        public string DiscType { get; set; } // DVD
        public string Director { get; set; } // DVD
        public int Runtime { get; set; } // DVD
        public string Studio { get; set; } // DVD
        public string Subtitles { get; set; } // DVD
        public string DvdGenre { get; set; } // DVD
                                             // ... (thêm các thuộc tính khác nếu cần)
    }
}
